singularity-eos/eos/modifiers/ramps_eos.hpp

```diff
@@ -0,0 +1,163 @@
+  }
+  PORTABLE_FUNCTION
+  Real PressureFromDensityInternalEnergy(const Real rho, const Real sie,
+                                         Real *lambda = nullptr) const {
+    // ramp pressure
+    const Real p_ramp {get_ramp_pressure(rho)};
+    // pressure from eos
+    const Real p_eos {t_.PressureFromDensityInternalEnergy(rho, sie, lambda)};
+    // return max(p_ramp, p_eos)
+    return p_eos < p_ramp ? p_ramp : p_eos;
+  }
+  PORTABLE_FUNCTION
+  Real SpecificHeatFromDensityInternalEnergy(const Real rho, const Real sie,
+                                             Real *lambda = nullptr) const {
+    return t_.SpecificHeatFromDensityInternalEnergy(rho, sie, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real BulkModulusFromDensityInternalEnergy(const Real rho, const Real sie,
+                                            Real *lambda = nullptr) const {
+    return t_.BulkModulusFromDensityInternalEnergy(rho, sie, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real GruneisenParamFromDensityInternalEnergy(const Real rho, const Real sie,
+                                               Real *lambda = nullptr) const {
+    return t_.GruneisenParamFromDensityInternalEnergy(rho, sie, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real PressureFromDensityTemperature(const Real rho, const Real temperature,
+                                      Real *lambda = nullptr) const {
+    const Real p_ramp {get_ramp_pressure(rho)};
+    const Real p_eos {t_.PressureFromDensityTemperature(rho, temperature, lambda)};
+    return p_eos < p_ramp ? p_ramp : p_eos;
+  }
+  PORTABLE_FUNCTION
+  Real SpecificHeatFromDensityTemperature(const Real rho, const Real temperature,
+                                          Real *lambda = nullptr) const {
+    return t_.SpecificHeatFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real BulkModulusFromDensityTemperature(const Real rho, const Real temperature,
+                                         Real *lambda = nullptr) const {
+    return t_.BulkModulusFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real GruneisenParamFromDensityTemperature(const Real rho, const Real temperature,
+                                            Real *lambda = nullptr) const {
+    return t_.GruneisenParamFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  void FillEos(Real &rho, Real &temp, Real &energy, Real &press, Real &cv, Real &bmod,
+               const unsigned long output, Real *lambda = nullptr) const {
+    Real senergy;
+    switch (t_.PreferredInput()) {
+    case thermalqs::density | thermalqs::temperature:
+      t_.FillEos(rho, temp, energy, press, cv, bmod, output, lambda);
+      energy = energy + shift_;
+      break;
+    case thermalqs::density | thermalqs::specific_internal_energy:
+      senergy = energy - shift_;
+      t_.FillEos(rho, temp, senergy, press, cv, bmod, output, lambda);
+      break;
```

Comment:
should we cover more cases?

Reply:
this is copy/paste code from shifted eos. I will work finishing updating the code when I get a chance.